### PR TITLE
Feature/ISSUE-02-Tabbar: Update Tababar

### LINKED ISSUE #7: [Feature] HomeTabbar/ Update UI
### Overview
- Create Tabbar

### Impact
-  Home screen

### Screenshots (Optional)
- // Capture affected screens

### PR DESCRIPTION
**Pull request by minh5896**
## Prerequisite
- [x] Self-review :monocle_face:
- [x] No Crashes ☠️
---

### Root cause
- // TabbarControler

### Solution
- // coordinator

### Impact
- // List of affected components/modules

### Screenshots (Optional)
- // Capture affected screen

s

### CODE REVIEW

File: MovieApp/Screens/Details/DetailsViewController.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+//  Created by MinhDev on 10/6/21.
+//
+
+import UIKit
+
+class DetailsViewController: UIViewController, Storyboarded {
+
+	override func viewDidLoad() {
+	super.viewDidLoad()
+	}
```
Comment (namnh92) [inline MovieApp/Screens/Details/DetailsViewController.swift:14]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Code4Fun-Group/Movie-App/blob/b9bc53c4ca90b499597b082b0a499ecbaeee8d75/MovieApp/Screens/Details/DetailsViewController.swift#L14"></span>Override methods which only call super can be removed</td>
    </tr>
  </tbody>
</table>

File: MovieApp/Screens/More/MoreViewController.swift [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+//  Created by MinhDev on 10/6/21.
+//
+
+import UIKit
+
+class MoreViewController: BaseViewController {
+
+	override func viewDidLoad() {
+		super.viewDidLoad()
+	}
```
Comment (namnh92) [inline MovieApp/Screens/More/MoreViewController.swift:14]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Code4Fun-Group/Movie-App/blob/b9bc53c4ca90b499597b082b0a499ecbaeee8d75/MovieApp/Screens/More/MoreViewController.swift#L14"></span>Override methods which only call super can be removed</td>
    </tr>
  </tbody>
</table>

**Comment by namnh92** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: MovieApp/Screens/Home/Views/HomeMovieCell/HomeMovieCellModel.swift [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ self.originalTitle = movie.originalTitle
+ self.overview = movie.overview
+ self.popularity = movie.popularity
+ self.posterPath = movie.posterPath
+ self.releaseDate = movie.releaseDate
+ self.title = movie.title
+ self.video = movie.video
+ self.voteAverage = movie.voteAverage
+ self.voteCount = movie.voteCount
+ }
```
Comment (namnh92) [inline MovieApp/Screens/Home/Views/HomeMovieCell/HomeMovieCellModel.swift]: @minh5896 : Add tabs
```suggestion
    var adult: Bool
    var backdropPath: String
    var genreIds: [Int]
    var id: Int
    var mediaType: String
    var originalLanguage: String
    var originalTitle: String
    var overview: String
    var popularity: Int
    var posterPath: String
    var releaseDate: String
    var title: String
    var video: Bool
    var voteAverage: Double
    var voteCount: Int

    init(_ movie: IMovieModel) {
        self.adult = movie.adult
        self.backdropPath = movie.backdropPath
        self.genreIds = movie.genreIds
        self.id = movie.id
        self.mediaType = movie.mediaType
        self.originalLanguage = movie.originalLanguage
        self.originalTitle = movie.originalTitle
        self.overview = movie.overview
        self.popularity = movie.popularity
        self.posterPath = movie.posterPath
        self.releaseDate = movie.releaseDate
        self.title = movie.title
        self.video = movie.video
        self.voteAverage = movie.voteAverage
        self.voteCount = movie.voteCount
    }
```

File: MovieApp/Screens/Home/Views/ListCell/ListMovieCell.swift [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+//  MovieApp
+//
+//  Created by MinhDev on 10/5/21.
+//
+
+import UIKit
+
+class ListMovieCell: UITableViewCell {
+	
+	@IBOutlet private weak var imageListMovieCell: UIImageView!
```
Comment (namnh92) [inline MovieApp/Screens/Home/Views/ListCell/ListMovieCell.swift:13]: @minh5896 : Add // MARK: -
```suggestion
        // MARK: - Outlets
	@IBOutlet private weak var imageListMovieCell: UIImageView!
```

File: MovieApp/Screens/Home/Views/ListCell/ListViewCell.swift [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+class ListViewCell: UITableViewCell {
+
+ override func awakeFromNib() {
+ super.awakeFromNib()
+ // Initialization code
+}
+override func setSelected(_ selected: Bool, animated: Bool)
+ super.setSelected(selected, animated: animated)
+// Configure the view for the selected state
+ }
```
Comment (namnh92) [inline MovieApp/Screens/Home/Views/ListCell/ListViewCell.swift:19]: @minh5896 : Add tabs

File: MovieApp/Screens/Home/Views/TVShowCell/TvShowViewCell.swift [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+import UIKit
+
+class TvShowViewCell: UITableViewCell {
+	// MARK: - IbOutlet
+	@IBOutlet private weak var titleTvShow: UILabel!
+	@IBOutlet private weak var imageTvShow: UIImageView!
+	@IBOutlet private weak var playTvShowButton: UIButton!
+	@IBOutlet private weak var myListTvShowButton: UIButton!
+	override func awakeFromNib() {
```
Comment (namnh92) [inline MovieApp/Screens/Home/Views/TVShowCell/TvShowViewCell.swift:17]: @minh5896 : Add space

File: MovieApp/Screens/More/MoreCoodinator.swift [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+//
+//  Created by MinhDev on 10/8/21.
+//
+
+import Foundation
+import UIKit
+
+class MoreCoordinator: CoordinatorProtocol {
+
+	var navigationController: UINavigationController?
```
Comment (namnh92) [inline MovieApp/Screens/More/MoreCoodinator.swift:13]:
```suggestion
       // MARK: - Variables
	var navigationController: UINavigationController?
```

File: MovieApp/Screens/More/ViewModel/File.swift [hunk 8]
```diff
@@ -0,0 +1,8 @@
+//
+//  File.swift
+//  MovieApp
+//
+//  Created by MinhDev on 10/6/21.
+//
+
+import Foundation
```
Comment (namnh92) [inline MovieApp/Screens/More/ViewModel/File.swift]: @minh5896 : Not use ?

File: MovieApp/Screens/More/MoreViewController.swift [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+//  MovieApp
+//
+//  Created by MinhDev on 10/6/21.
+//
+
+import UIKit
+
+class MoreViewController: BaseViewController {
+
+}
```
Comment (namnh92) [inline MovieApp/Screens/More/MoreViewController.swift:15]: @minh5896 : Remove not use ?

File: MovieApp/Screens/More/ViewModel/MoreViewModel.swift [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+//  MovieApp
+//
+//  Created by MinhDev on 10/6/21.
+//
+
+import Foundation
+protocol IMoreViewModel {
+	func getMovies()
+}
+class MoreViewModel {
```
Comment (namnh92) [inline MovieApp/Screens/More/ViewModel/MoreViewModel.swift]:
```suggestion
}

class MoreViewModel {
```

File: MovieApp/Screens/More/ViewModel/MoreViewModel.swift [hunk 11]
```diff
@@ -0,0 +1,20 @@
+//
+//  MoreViewModel.swift
+//  MovieApp
+//
+//  Created by MinhDev on 10/6/21.
+//
+
+import Foundation
+protocol IMoreViewModel {
```
Comment (namnh92) [inline MovieApp/Screens/More/ViewModel/MoreViewModel.swift]:
```suggestion
import Foundation

protocol IMoreViewModel {
```

File: MovieApp/Screens/Search/SearchCoodinator.swift [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+		self.navigationController = navigationController
+		self.navigationController?.isNavigationBarHidden = true
+
+		let view = SearchViewController.fromStoryboard()
+		view.coordinator = self
+		view.tabBarItem.image = UIImage(named: Icon.search)
+		view.tabBarItem.title = GlobalConstants.searchMovies
+		self.navigationController?.viewControllers = [view]
+
+	}
```
Comment (namnh92) [inline MovieApp/Screens/Search/SearchCoodinator.swift]:
```suggestion
		self.navigationController?.viewControllers = [view]
	}
```

File: MovieApp/SharedData/GlobalConstants.swift [hunk 13]
```diff
@@ -6,7 +6,21 @@
 //
 
 import Foundation
+import UIKit
```
Comment (namnh92) [inline MovieApp/SharedData/GlobalConstants.swift]:
```suggestion
```

**Review by namnh92 — CHANGES_REQUESTED** [review]

File: MovieApp/Coodinator/AppCoordinator.swift [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+	}
+
+	func searchViewController() {
+		let view = SearchViewController.fromStoryboard()
+		view.coordinator = self
+		navigationController?.setViewControllers([view], animated: true)
+
+		navigationController?.isNavigationBarHidden = true
+	}
+	func comingViewController() {
```
Comment (namnh92) [inline MovieApp/Coodinator/AppCoordinator.swift]:
```suggestion
	}
	
	func comingViewController() {
```

File: MovieApp/Coodinator/AppCoordinator.swift [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+		let view = ComingSoonViewController.fromStoryboard()
+		view.coordinator = self
+		navigationController?.setViewControllers([view], animated: true)
+
+		navigationController?.isNavigationBarHidden = true
+	}
+	func downloadViewController() {
```
Comment (namnh92) [inline MovieApp/Coodinator/AppCoordinator.swift]:
```suggestion
	}
	
	func downloadViewController() {
```

File: MovieApp/Coodinator/CoordinatorDelegate.swift [hunk 16]
```diff
@@ -0,0 +1,14 @@
+//
+//  Coordinator.swift
+//  MovieApp
+//
+//  Created by MinhDev on 10/7/21.
+//
+
+import Foundation
```
Comment (namnh92) [inline MovieApp/Coodinator/CoordinatorDelegate.swift]:
```suggestion
```

File: MovieApp/Coodinator/CoordinatorProtocol.swift [hunk 17]
```diff
@@ -0,0 +1,31 @@
+//
+//  AppCoordinator.swift
+//  MovieApp
+//
+//  Created by MinhDev on 10/7/21.
+//
+
+import Foundation
```
Comment (namnh92) [inline MovieApp/Coodinator/CoordinatorProtocol.swift]:
```suggestion
```

File: MovieApp/Networking/Entities/Movie/MovieResponse.swift [hunk 18]
```diff
@@ -2,49 +2,64 @@
...
+	case originalLanguage = "original_language"
+	case originalTitle = "original_title"
+	case overview = "overview"
+	case popularity = "popularity"
+	case posterPath = "poster_path"
+	case releaseDate = "release_date"
+	case title = "title"
+	case video = "video"
+	case voteAverage = "vote_average"
+	case voteCount = "vote_count"
```
Comment (namnh92) [inline MovieApp/Networking/Entities/Movie/MovieResponse.swift]:
```suggestion
	
	enum CodingKeys: String, CodingKey {
		case adult = "adult"
		case backdropPath = "backdrop_path"
		case genreIds = "genre_ids"
		case id = "id"
		case mediaType = "media_type"
		case originalLanguage = "original_language"
		case originalTitle = "original_title"
		case overview = "overview"
		case popularity = "popularity"
		case posterPath = "poster_path"
		case releaseDate = "release_date"
		case title = "title"
		case video = "video"
		case voteAverage = "vote_average"
		case voteCount = "vote_count"
	}
```

**Review by namnh92 — CHANGES_REQUESTED** [review]

File: MovieApp/Screens/Home/Views/TVShowCell/TvShowViewCell.swift (hunk 6 above)
Comment (minh5896) [reply MovieApp/Screens/Home/Views/TVShowCell/TvShowViewCell.swift:17]: fix complete

File: MovieApp/Screens/More/MoreCoodinator.swift (hunk 7 above)
Comment (minh5896) [reply MovieApp/Screens/More/MoreCoodinator.swift:13]: fix complete

File: MovieApp/Screens/More/ViewModel/File.swift (hunk 8 above)
Comment (minh5896) [reply MovieApp/Screens/More/ViewModel/File.swift]: not use. deleted

File: MovieApp/Screens/More/MoreViewController.swift (hunk 9 above)
Comment (minh5896) [reply MovieApp/Screens/More/MoreViewController.swift:15]: fix complete

File: MovieApp/Screens/More/ViewModel/MoreViewModel.swift (hunk 10 above)
Comment (minh5896) [reply MovieApp/Screens/More/ViewModel/MoreViewModel.swift]: fix complete

File: MovieApp/Screens/More/ViewModel/MoreViewModel.swift (hunk 11 above)
Comment (minh5896) [reply MovieApp/Screens/More/ViewModel/MoreViewModel.swift]: fix complete

File: MovieApp/Screens/Search/SearchCoodinator.swift (hunk 12 above)
Comment (minh5896) [reply MovieApp/Screens/Search/SearchCoodinator.swift]: fix complete

File: MovieApp/SharedData/GlobalConstants.swift (hunk 13 above)
Comment (minh5896) [reply MovieApp/SharedData/GlobalConstants.swift]: deleted

File: MovieApp/Coodinator/AppCoordinator.swift (hunk 14 above)
Comment (minh5896) [reply MovieApp/Coodinator/AppCoordinator.swift]: fix completed

File: MovieApp/Coodinator/AppCoordinator.swift (hunk 15 above)
Comment (minh5896) [reply MovieApp/Coodinator/AppCoordinator.swift]: fix complete

File: MovieApp/Coodinator/CoordinatorDelegate.swift (hunk 16 above)
Comment (minh5896) [reply MovieApp/Coodinator/CoordinatorDelegate.swift]: fix complete

File: MovieApp/Coodinator/CoordinatorProtocol.swift (hunk 17 above)
Comment (minh5896) [reply MovieApp/Coodinator/CoordinatorProtocol.swift]: fix complete

File: MovieApp/Networking/Entities/Movie/MovieResponse.swift (hunk 18 above)
Comment (minh5896) [reply MovieApp/Networking/Entities/Movie/MovieResponse.swift]: fix completed

**Pull request merged**